### PR TITLE
chore: add send_action method to persistence task

### PR DESCRIPTION
**Pull request by Rjected**
This method lets the caller determine if they want to await on the result of the action

### CODE REVIEW

File: crates/engine/tree/src/persistence.rs [hunk 1]
```diff
@@ -148,23 +148,31 @@ where
             match action {
                 PersistenceAction::RemoveBlocksAbove((new_tip_num, sender)) => {
                     let output = self.remove_blocks_above(new_tip_num);
-                    sender.send(output).unwrap();
+
+                    // we ignore the error because the caller may or may not care about the result
+                    let _ = sender.send(output);
```
Comment (fgimenez) [inline crates/engine/tree/src/persistence.rs:153]: Maybe too early to focus on this, but could we log the potential error here?

File: crates/engine/tree/src/persistence.rs [hunk 2]
```diff
@@ -203,6 +211,12 @@ impl PersistenceHandle {
         Self { sender }
     }
 
+    /// Sends a specific [`PersistenceAction`] in the contained channel. The caller is responsible
+    /// for creating any channels for the given action.
+    pub fn send_action(&self, action: PersistenceAction) {
+        self.sender.send(action).expect("should be able to send");
```
Comment (fgimenez) [inline crates/engine/tree/src/persistence.rs:217]: Could we prevent this panic? If the caller should take care of creating the channels then this is open to failures from outside?

**Review by fgimenez — APPROVED** [review]
Looks good!

File: crates/engine/tree/src/persistence.rs (hunk 2 above)
Comment (Rjected) [reply crates/engine/tree/src/persistence.rs:217]: Yeah, we could bubble the error

File: crates/engine/tree/src/persistence.rs (hunk 1 above)
Comment (Rjected) [reply crates/engine/tree/src/persistence.rs:153]: The error returned here is not actually an error, instead it's the item itself if sending is not possible:
https://docs.rs/tokio/latest/tokio/sync/oneshot/struct.Sender.html#method.send

> An unsuccessful send would be one where the corresponding receiver has already been deallocated

So there is only really error case, which does not necessarily indicate a failure mode

**Pull request merged**